Dapper/SqlMapper.cs

```diff
@@ -2532,11 +2532,13 @@ internal static IList<LiteralToken> GetLiteralTokens(string sql)
                 throw new NotSupportedException("ValueTuple should not be used for parameters - the language-level names are not available to use as parameter names, and it adds unnecessary boxing");
             }
 
-            bool filterParams = false;
-            if (removeUnused && identity.commandType.GetValueOrDefault(CommandType.Text) == CommandType.Text)
+            bool filterParams = removeUnused && identity.commandType.GetValueOrDefault(CommandType.Text) == CommandType.Text;
+            
+            if (Settings.SupportLegacyParameterTokens && filterParams)

```

Comment:
nit; can we check the local first?

Reply:
Sorry, I'm not sure I follow. Check what?

Reply:
Swap the order of the two operands -`if (filterParams && ...)`

Reply:
Sure, will do. Out of curiosity, why does it matter?

Reply:
it isn't huge, hence "nit" - but checking a local in the current stack-frame is faster than pretty much anything else; now, compared to going out-of-process and probably off-box for the actual SQL: this is nothing - we've spent more time talking about it here than will *ever* be saved by the change, but... meh, I guess I just prefer it that way around, despite there not being an objectively demonstrable "this will genuinely help" :p